Gems/SceneProcessing/Code/Source/Generation/Components/TangentGenerator/TangentGenerateComponent.cpp

```diff
@@ -191,95 +184,175 @@ namespace AZ::SceneGenerationComponents
+                    // In case there are no tangents/bitangents while the user selected to use the source ones, default to MikkT.
+                    AZ_TracePrintf(AZ::SceneAPI::Utilities::WarningWindow, "Cannot use source scene tangents as there are none in the asset for mesh '%s' for uv set %zu. Defaulting to generating tangents using MikkT.\n",
+                        scene.GetGraph().GetNodeName(nodeIndex).GetName(), uvSetIndex);
+                    tangentSpace = AZ::SceneAPI::DataTypes::TangentSpace::MikkT;
                 }
-                break;
+            }
 
-                // Generate using MikkT space.
-                case AZ::SceneAPI::DataTypes::TangentSpace::MikkT:
+            if (!tangentData)
+            {
+                if (!AZ::SceneGenerationComponents::TangentGenerateComponent::CreateTangentLayer(scene.GetManifest(), nodeIndex, meshData->GetVertexCount(), uvSetIndex,
+                    tangentSpace, graph, &tangentData))
                 {
-                    allSuccess &= AZ::TangentGeneration::Mesh::MikkT::GenerateTangents(
-                        scene.GetManifest(), graph, nodeIndex, const_cast<AZ::SceneAPI::DataTypes::IMeshData*>(meshData), uvSetIndex);
-                    for (AZ::SceneData::GraphData::BlendShapeData* blendShape : blendShapes)
-                    {
-                        allSuccess &= AZ::TangentGeneration::BlendShape::MikkT::GenerateTangents(blendShape, uvSetIndex);
-                    }
+                    AZ_TracePrintf(AZ::SceneAPI::Utilities::ErrorWindow, "Failed to create tangents data set for mesh %s for uv set %zu.\n",
+                        scene.GetGraph().GetNodeName(nodeIndex).GetName(), uvSetIndex);
+                    continue;
+                }
+            }
+            if (!bitangentData)
+            {
+                if (!AZ::SceneGenerationComponents::TangentGenerateComponent::CreateBitangentLayer(scene.GetManifest(), nodeIndex, meshData->GetVertexCount(), uvSetIndex,
+                    tangentSpace, graph, &bitangentData))
+                {
+                    AZ_TracePrintf(AZ::SceneAPI::Utilities::ErrorWindow, "Failed to create bitangents data set for mesh %s for uv set %zu.\n",
+                        scene.GetGraph().GetNodeName(nodeIndex).GetName(), uvSetIndex);
+                    continue;
                 }
-                break;
+            }
+            tangentData->SetTangentSpace(tangentSpace);
+            bitangentData->SetTangentSpace(tangentSpace);
 
-                // If we use EMotion FX calculated tangents, we don't need to generate this here.
-                case AZ::SceneAPI::DataTypes::TangentSpace::EMotionFX:
-                    allSuccess &= true;
-                    break;
+            switch (tangentSpace)
+            {
+            // If we want source scene tangents, we don't need to do anything for that.
+            case AZ::SceneAPI::DataTypes::TangentSpace::FromSourceScene:
```

Comment:
Looking at the code above, this probably is unachievable code? Because had the user select the fromSourceScene option, either it will hit continue or it will be forced to use MikkT. I think when code reach this point the space option will always be mikkt.
We can keep the default check here to other tagent space option added.

Reply:
Good spot, removed